tests/reproducibility/test_attri2vec.py

```diff
@@ -0,0 +1,151 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import numpy as np
+import pytest
+import random
+import tensorflow as tf
+from stellargraph.data.unsupervised_sampler import UnsupervisedSampler
+from stellargraph.mapper.sampled_link_generators import Attri2VecLinkGenerator
+from stellargraph.layer.attri2vec import Attri2Vec
+from stellargraph.layer.link_inference import link_classification
+from stellargraph.random import set_seed
+from ..test_utils.graphs import petersen_graph
+from .fixtures import assert_reproducible
+
+
+def unsup_attri2vec_model(generator, optimizer, bias, normalize):
+    layer_sizes = [50]
+    attri2vec = Attri2Vec(
+        layer_sizes=layer_sizes, generator=generator, bias=bias, normalize=normalize,
+    )
+    # Build the model and expose input and output sockets of attri2vec, for node pair inputs:
+    x_inp, x_out = attri2vec.build()
+    prediction = link_classification(
+        output_dim=1, output_act="sigmoid", edge_embedding_method="ip"
+    )(x_out)
+    model = tf.keras.Model(inputs=x_inp, outputs=prediction)
+
+    model.compile(optimizer=optimizer, loss=tf.keras.losses.binary_crossentropy)
+
+    return model
+
+
+def unsup_attri2vec(
+    g,
+    optimizer,
+    batch_size=4,
+    epochs=4,
+    bias=True,
+    normalize="l2",
+    number_of_walks=1,
+    walk_length=5,
+    seed=0,
+    shuffle=True,
+):
+    set_seed(seed)
+    tf.random.set_seed(seed)
+    if shuffle:
+        random.seed(seed)
+
+    nodes = list(g.nodes())
+    unsupervised_samples = UnsupervisedSampler(
+        g, nodes=nodes, length=walk_length, number_of_walks=number_of_walks
+    )
+    generator = Attri2VecLinkGenerator(g, batch_size)
+    train_gen = generator.flow(unsupervised_samples)
+
+    model = unsup_attri2vec_model(generator, optimizer, bias, normalize)
+
+    model.fit_generator(
+        train_gen, epochs=epochs, use_multiprocessing=False, workers=4, shuffle=shuffle,
+    )
+    return model
+
+
+def attri2vec_link_pred_model(generator, optimizer, bias, normalize):
+    layer_sizes = [50]
+    attri2vec = Attri2Vec(
+        layer_sizes=layer_sizes, generator=generator, bias=bias, normalize=normalize,
+    )
+    # Build the model and expose input and output sockets of attri2vec, for node pair inputs:
+    x_inp, x_out = attri2vec.build()
+    pred = link_classification(
+        output_dim=1, output_act="relu", edge_embedding_method="ip"
+    )(x_out)
+
+    model = tf.keras.Model(inputs=x_inp, outputs=pred)
+
+    model.compile(
+        optimizer=optimizer, loss=tf.keras.losses.binary_crossentropy,
+    )
+
+    return model
+
+
+def attri2vec_link_prediction(
+    g,
+    edge_ids,
+    edge_labels,
+    optimizer,
```

Comment:
Out of curiosity, what's the purpose of this parameter? It seems like it's always `tf.optimizers.Adam(1e-3)`?

Similarly, why have the other configurations (other than `shuffle`)? It seems like they're never customised?

Reply:
hmm yeah that is very true 😅 

Maybe it makes more sense to have:
```python
def attri2vec_link_prediction(g, edge_ids, edge_labels, batch_size, shuffle)
```
We don't actually need any optional parameters